synapse/handlers/message.py

```diff
@@ -999,7 +999,26 @@ async def create_and_send_nonmember_event(
             raise ShadowBanError()
 
         if ratelimit:
-            await self.request_ratelimiter.ratelimit(requester, update=False)
+            room_id = event_dict["room_id"]
+            try:
+                room_version = await self.store.get_room_version(room_id)
+            except NotFoundError:
+                # If the room doesnt' exist.
+                raise AuthError(403, f"User {requester.user} not in room {room_id}")
```

Comment:
Is this the right error message?

Reply:
It matches the current error message I believe. There's probably some argument about not wanting to leak the existence of rooms or something :shrug: